src/Provider/AbstractProvider.php

```diff
@@ -364,21 +368,14 @@ public function getAuthenticatedRequest($method, $url, AccessToken $token)
     public function getResponse(RequestInterface $request)
     {
         try {
-            $client = $this->getHttpClient();
-
-            $httpResponse = $client->sendRequest($request);
-
-            $response = (string) $httpResponse->getBody();
-        } catch (HttpAdapterException $e) {
-            if (!$e->hasResponse()) {
-                throw $e;
-            }
-            $response = (string) $e->getResponse()->getBody();
+            $response = $this->getHttpClient()->send($request);
```

Comment:
And here we have a very convenient method to send a request and get back a response. It's currently backed by Guzzle, but that is completely hidden from the end user.